internal/services/mongocluster/mongo_cluster_user_resource_test.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type MongoClusterUserResource struct{}
+
+func TestAccMongoClusterUser_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_user", "test")
+	r := MongoClusterUserResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMongoClusterUser_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_user", "test")
+	r := MongoClusterUserResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccMongoClusterUser_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_user", "test")
+	r := MongoClusterUserResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r MongoClusterUserResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := users.ParseUserID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.MongoCluster.UsersClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r MongoClusterUserResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_mongo_cluster_user" "test" {
+  object_id              = data.azurerm_client_config.current.object_id
+  mongo_cluster_id       = azurerm_mongo_cluster.test.id
+  identity_provider_type = "MicrosoftEntraID"
+  principal_type         = "servicePrincipal"
+
+  role {
+    database = "admin"
+    role     = "root"
+  }
+}
+`, r.template(data))
+}
+
+func (r MongoClusterUserResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_mongo_cluster_user" "import" {
+  object_id              = azurerm_mongo_cluster_user.test.object_id
+  mongo_cluster_id       = azurerm_mongo_cluster_user.test.mongo_cluster_id
+  identity_provider_type = azurerm_mongo_cluster_user.test.identity_provider_type
+  principal_type         = azurerm_mongo_cluster_user.test.principal_type
+
+  role {
+    database = azurerm_mongo_cluster_user.test.role.0.database
+    role     = azurerm_mongo_cluster_user.test.role.0.role
+  }
+}
+`, r.basic(data))
+}
+
+func (r MongoClusterUserResource) complete(data acceptance.TestData) string {
```

Comment:
since all properties are `Required`, what's the difference between `basic` and `complete` testcase?

Reply:
the `complete` test is renamed as `principalTypeUser` test